packages/next/src/build/index.ts

```diff
@@ -1258,7 +1259,7 @@ export default async function build(
               header: RSC_HEADER,
               // This vary header is used as a default. It is technically re-assigned in `base-server`,
               // and may include an additional Vary option for `Next-URL`.
-              varyHeader: `${RSC_HEADER}, ${NEXT_ROUTER_STATE_TREE_HEADER}, ${NEXT_ROUTER_PREFETCH_HEADER}`,
+              varyHeader: `${RSC_HEADER}, ${NEXT_ROUTER_STATE_TREE_HEADER}, ${NEXT_ROUTER_PREFETCH_HEADER}, ${NEXT_ROUTER_SEGMENT_PREFETCH_HEADER}`,
```

Comment:
I think we'll need to duplicate this in [fetchServerResponse](https://github.com/vercel/next.js/blob/e2fb733e25b3c36d2b572439a9efb63a58dfbfb0/packages/next/src/client/components/router-reducer/fetch-server-response.ts#L132-L138) to mirror the workaround we have for CDNs that don't respect `Vary` (the computed `?_rsc` param)

Reply:
Ah right, forgot about that one, will do

Reply:
Actually I remember now: I left that alone for now since the client doesn't actually issue any of these requests yet. So it doesn't matter. Will update once I update the client.